C++/API/API.cpp

```diff
@@ -499,87 +452,81 @@ bool iSENSE::get_datasets_and_mediaobjects()
-    // get the project's fields.
-    if(http_code != 200)
-    {
-      cerr << "\nGET project fields failed.\n";
-      cerr << "Is the project ID you entered valid?\n";
+  /*  The iSENSE API gives us one response code to check against:
+   *  Success: 200 OK
+   *  Failure: 404 Not Found
+   */
 
-      // Clean up cURL and close the memfile
-      curl_easy_cleanup(curl);
-      curl_global_cleanup();
-      memfclose(json_file);
+  // If we do not get a code 200, or cURL quits for some reason,
+  // we didn't successfully get the project's fields.
+  if (http_code != 200) {
+    std::cerr << "\nError in: get_datasets_and_mediaobjects().\n";
+    std::cerr << "GET project fields failed.\n";
+    std::cerr << "Is the project ID you entered valid?\n";
 
-      return false;
-    }
+    // Close the memfile
+    memfclose(json_file);
 
-    // We got a code 200, so try and parse the JSON into a PICOJSON object.
-    // Error checking for the parsing occurs below.
-    string errors;
+    return false;
+  }
 
-    // This will parse the JSON file.
-    parse(get_data, json_file->data, json_file->data + json_file->size, &errors);
+  // We got a code 200, so try and parse the JSON into a PICOJSON object.
+  // Error checking for the parsing occurs below.
+  std::string errors;
 
-    // If we have errors, print them out and quit.
-    if(errors.empty() != true)
-    {
-      cerr << "\nError parsing JSON file in method: get_datasets_and_mediaobjects()\n";
-      cerr << "Error was: " << errors;
-      return false;
-    }
+  // This will parse the JSON file.
+  parse(get_data, json_file->data, json_file->data + json_file->size, &errors);
 
-    // Save the fields to the field array
-    fields = get_data.get("fields");
-    fields_array = fields.get<array>();
+  // close the memfile
+  memfclose(json_file);
 
-    // Save the datasets to the datasets array
-    value temp = get_data.get("dataSets");
-    data_sets = temp.get<array>();
+  // If we have errors, print them out and quit.
+  if (errors.empty() != true) {
+    std::cerr << "\nError in method: get_datasets_and_mediaobjects()\n";
+    std::cerr << "Parsing JSON file failed. Error was: " << errors;
+    return false;
+  }
 
-    // Save the media objects to the media objects array
-    value temp2 = get_data.get("mediaObjects");
-    media_objects = temp2.get<array>();
+  // Save the fields to the field array
+  fields = get_data.get("fields");
+  fields_array = fields.get<array>();
 
-    // Save the owner info.
-    value temp3 = get_data.get("owner");
-    owner_info = temp3.get<object>();
-  }
+  // Save the datasets to the datasets array
+  value temp = get_data.get("dataSets");
+  data_sets = temp.get<array>();
 
-  // Clean up cURL and close the memfile
-  curl_easy_cleanup(curl);
-  curl_global_cleanup();
-  memfclose(json_file);
+  // Save the media objects to the media objects array
+  value temp2 = get_data.get("mediaObjects");

```

Comment:
You no longer need `temp` once you get to this point, so why not just reuse `temp` instead of defining a `temp2` and `temp3`?